### PR TITLE
WAVE2 926 FGS Supervisor admin selecting degrees

### PR DESCRIPTION
**Pull request by robbdavis**
WAVE2 926 FGS Supervisor admin selecting degrees

### CODE REVIEW

File: sites/all/modules/ucalgary/ucgy_fgs/ucgy_fgs.features.field_base.inc [hunk 1]
```diff
@@ -138,8 +138,8 @@ function ucgy_fgs_field_default_field_bases() {
     'module' => 'list',
     'settings' => array(
       'allowed_values' => array(
-        'thesis-based' => 'Thesis based',
-        'course-based' => 'Course based',
+        'Thesis based ' => ' Thesis based',
```
Comment (balrajb) [inline sites/all/modules/ucalgary/ucgy_fgs/ucgy_fgs.features.field_base.inc]: can we make sure, we don't use spaces in key! the explore program filter wouldn't work.

**Comment by robbdavis** [comment]
Balraj I made those changes to the keys. Those changes are not yet on the fgs branch. Once this gets merged, your changes will be on there.

File: sites/all/modules/ucalgary/ucgy_gbl/ucgy_gbl_supervisor/ucgy_gbl_supervisor.module [hunk 2]
```diff
@@ -7,6 +7,21 @@
 include_once 'ucgy_gbl_supervisor.features.inc';
 
 /**
+ * Implements hook_views_api().
+ *
+ * Define Views API version.
+ */
+function ucgy_gbl_supervisor_views_api()
```
Comment (gsowrey) [inline sites/all/modules/ucalgary/ucgy_gbl/ucgy_gbl_supervisor/ucgy_gbl_supervisor.module]: Please make sure this isn't in the Feature export. It has a way of always coming back...

File: sites/all/modules/ucalgary/ucgy_gbl/ucgy_gbl_supervisor/ucgy_gbl_supervisor.module [hunk 3]
```diff
@@ -34,9 +49,15 @@ function ucgy_gbl_supervisor_form_alter(&$form, &$form_state, $form_id) {
...
 
+//        // Attach jquery - leaving this in here for future js filter improvement
+//        $form['#attached']['js'] = array(
+//           drupal_get_path('module', 'ucgy_gbl_supervisor') . '/js/ucgy_gbl_supervisor.js',
+//        );
+
         //  Hide the fgs degree programs list if ucgy_fgs module  is not enabled.
         if (!module_exists('ucgy_fgs')) {
             hide($form['field_gbl_supervisor_degr_superv']);
+            hide($form['field_gbl_supervisor_dgre_superv']);
```
Comment (gsowrey) [inline sites/all/modules/ucalgary/ucgy_gbl/ucgy_gbl_supervisor/ucgy_gbl_supervisor.module]: Uhhhh... these are named awfully similarly? Are they different?

File: sites/all/modules/ucalgary/ucgy_gbl/ucgy_gbl_supervisor/ucgy_gbl_supervisor.module [hunk 4]
```diff
@@ -58,6 +79,18 @@ function ucgy_gbl_supervisor_form_alter(&$form, &$form_state, $form_id) {
             'data' => '.form-item-field-gbl-supervisor-show-form-und { padding-top: 40px !important; } fieldset { padding-top: 10px !important; }',
             'type' => 'inline',
         );
+
     }
-}
 
+    if($form['#id'] == 'views-exposed-form-fgs-supervisor-degrees-supervised-entityreference-view-widget-1') {
```
Comment (gsowrey) [inline sites/all/modules/ucalgary/ucgy_gbl/ucgy_gbl_supervisor/ucgy_gbl_supervisor.module]: Is this different than the $form_id value?

File: sites/all/modules/ucalgary/ucgy_gbl/ucgy_gbl_supervisor/ucgy_gbl_supervisor.module (hunk 3 above)
Comment (robbdavis) [inline sites/all/modules/ucalgary/ucgy_gbl/ucgy_gbl_supervisor/ucgy_gbl_supervisor.module]: Removed the one not needed any more.

I thought I had removed it in this branch already. I guess not. Good catch.

File: sites/all/modules/ucalgary/ucgy_gbl/ucgy_gbl_supervisor/ucgy_gbl_supervisor.module (hunk 2 above)
Comment (robbdavis) [inline sites/all/modules/ucalgary/ucgy_gbl/ucgy_gbl_supervisor/ucgy_gbl_supervisor.module]: I hear ya. Only one call to this from .module.

File: sites/all/modules/ucalgary/ucgy_gbl/ucgy_gbl_supervisor/ucgy_gbl_supervisor.info [hunk 5]
```diff
@@ -62,6 +66,7 @@ features[field_instance][] = node-gbl_supervisor-field_gbl_supervisor_available
 features[field_instance][] = node-gbl_supervisor-field_gbl_supervisor_cred_abrv
 features[field_instance][] = node-gbl_supervisor-field_gbl_supervisor_credentials
 features[field_instance][] = node-gbl_supervisor-field_gbl_supervisor_degr_superv
```
Comment (balrajb) [inline sites/all/modules/ucalgary/ucgy_gbl/ucgy_gbl_supervisor/ucgy_gbl_supervisor.info]: should remove `field_gbl_supervisor_degr_superv` as we have `field_gbl_supervisor_dgre_superv`

File: sites/all/modules/ucalgary/ucgy_gbl/ucgy_gbl_supervisor/ucgy_gbl_supervisor.info [hunk 6]
```diff
@@ -36,6 +39,7 @@ features[field_base][] = field_gbl_supervisor_credentials
 features[field_base][] = field_gbl_supervisor_degr_superv
```
Comment (balrajb) [inline sites/all/modules/ucalgary/ucgy_gbl/ucgy_gbl_supervisor/ucgy_gbl_supervisor.info]: should remove `field_gbl_supervisor_degr_superv` as we have `field_gbl_supervisor_dgre_superv`

File: sites/all/modules/ucalgary/ucgy_gbl/ucgy_gbl_supervisor/ucgy_gbl_supervisor.install [hunk 7]
```diff
@@ -28,3 +28,20 @@ function ucgy_gbl_supervisor_uninstall() {
...
 }
+
+/**
+ * Implements hook_update().
+ * Deleting fields that are no longer needed.
+ */
+function ucgy_gbl_supervisor_update_0001()
+{
+$fields = array(
+    'field_gbl_supervisor_degr_superv',
```
Comment (balrajb) [inline sites/all/modules/ucalgary/ucgy_gbl/ucgy_gbl_supervisor/ucgy_gbl_supervisor.install]: should use `field_gbl_supervisor_dgre_superv` , right?

File: sites/all/modules/ucalgary/ucgy_gbl/ucgy_gbl_supervisor/ucgy_gbl_supervisor.module (hunk 4 above)
Comment (robbdavis) [inline sites/all/modules/ucalgary/ucgy_gbl/ucgy_gbl_supervisor/ucgy_gbl_supervisor.module]: I'm not sure why, but $form_id doesn't work here; it throws an ajax error when I click the reset button. As is, ($form['#id']), it works.

File: sites/all/modules/ucalgary/ucgy_gbl/ucgy_gbl_supervisor/ucgy_gbl_supervisor.install (hunk 7 above)
Comment (robbdavis) [inline sites/all/modules/ucalgary/ucgy_gbl/ucgy_gbl_supervisor/ucgy_gbl_supervisor.install]: As is is correct, we are deleting field_gbl_supervisor_degr_superv and keeping field_gbl_supervisor_dgre_superv

File: sites/all/modules/ucalgary/ucgy_gbl/ucgy_gbl_supervisor/ucgy_gbl_supervisor.info (hunk 6 above)
Comment (robbdavis) [inline sites/all/modules/ucalgary/ucgy_gbl/ucgy_gbl_supervisor/ucgy_gbl_supervisor.info]: Really good catch Balraj. This must have been pulled back in when I merged fgs to resolve conflicts. I'll fix this now and comment when fix is up.

File: sites/all/modules/ucalgary/ucgy_gbl/ucgy_gbl_supervisor/ucgy_gbl_supervisor.info (hunk 5 above)
Comment (robbdavis) [inline sites/all/modules/ucalgary/ucgy_gbl/ucgy_gbl_supervisor/ucgy_gbl_supervisor.info]: Good catch balraj. These must have been put back in when I merged fgs to resolve some conflicts. All references to field_gbl_supervisor_degr_superv have been removed except in the .install file where we are actually deleting this field.

**Comment by robbdavis** [comment]
Last commit should address all these issues. Thanks guys. Let me know if there is anything else.

**Pull request merged**